raiden/network/proxies/token.py

```diff
@@ -99,49 +101,30 @@ def approve(self, allowed_address: Address, allowance: TokenAmount) -> None:
                         block_identifier=block,
                     )
 
-                    msg = self._check_why_approved_failed(allowance, block)
-                    raise RaidenUnrecoverableError(f"{error_prefix}. {msg}")
-
-    def _check_why_approved_failed(
-        self, allowance: TokenAmount, block_identifier: BlockSpecification
-    ) -> str:
-        user_balance = self.balance_of(
-            address=self.client.address, block_identifier=block_identifier
-        )
-
-        # If the balance is zero, either the smart contract doesnt have a
-        # balanceOf function or the actual balance is zero
-        if user_balance == 0:
-            msg = (
-                "Approve failed. \n"
-                "Your account balance is 0 (zero), either the smart "
-                "contract is not a valid ERC20 token or you don't have funds "
-                "to use for openning a channel. "
-            )
-
-        # The approve call failed, check the user has enough balance
-        # (assuming the token smart contract may check for the maximum
-        # allowance, which is not necessarily the case)
-        elif user_balance < allowance:
-            msg = (
-                f"Approve failed. \n"
-                f"Your account balance is {user_balance}. "
-                f"The requested allowance is {allowance}. "
-                f"The smart contract may be rejecting your request due to the "
-                f"lack of balance."
-            )
-
-        # If the user has enough balance, warn the user the smart contract
-        # may not have the approve function.
-        else:
-            msg = (
-                f"Approve failed. \n"
-                f"Your account balance is {user_balance}. Nevertheless the call to"
-                f"approve failed. Please make sure the corresponding smart "
-                f"contract is a valid ERC20 token."
-            ).format(user_balance)
-
-        return msg
+                    self._raise_if_token_balance_insufficient(allowance, prefix=error_prefix)
```

Comment:
This line says balance **is** a pre-condition. So this should be checked before. However, reusing the functionality from the function below is wrong.

Reply:
Thanks for the quick review!

As I understand it there are no preconditions in this case, as `Token` is only an interface, not an actual raiden contract. 

We could check for balance as a precondition, but that is checked in the proxies of the actual raiden contracts calling into `Token`, i. e. `TokenNetwork` and `UserDeposit`. 
Checking the balance precondition here would be just redundant, and taking the check out of the other proxies would be messy (preconditions should be checked before anything else happens). So I just added to the docstrings that these functions expect the balance precondition to be checked.